From 5b7a90ffad43927db6851e1f953d21727aa29ea2 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Tue, 2 Dec 2014 11:33:15 +0000
Subject: [PATCH 07/10] Remove redundant re-checking of return from read_msr()

The re-checking of the read_msr return code is redundant
and should be removed.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_msr.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/thd_msr.cpp b/src/thd_msr.cpp
index 218f105..c003f7c 100644
--- a/src/thd_msr.cpp
+++ b/src/thd_msr.cpp
@@ -469,8 +469,6 @@ int cthd_msr::inc_freq_state() {
 		if (ret < 0)
 			return THD_ERROR;
 		thd_log_debug("perf_ctl current %x\n", (unsigned int) val);
-		if (ret < 0)
-			return THD_ERROR;
 
 		current_clock = (val >> 8) & 0xff;
 		current_clock++;
-- 
2.1.3

